src/Gui/Action.cpp

```diff
@@ -921,15 +921,26 @@ void RecentMacrosAction::setFiles(const QStringList& files)
     int numRecentFiles = std::min<int>(recentFiles.count(), files.count());
     for (int index = 0; index < numRecentFiles; index++) {
         QFileInfo fi(files[index]);
-        QString accel = QString::fromStdString(shortcut_modifiers);
-        accel.append(QString::number(index+1,10)).toStdString();
         recentFiles[index]->setText(QString::fromLatin1("%1 %2").arg(index+1).arg(fi.baseName()));
-        recentFiles[index]->setStatusTip(tr("Run macro %1 (Shift+click to edit) shortcut: %2").arg(files[index]).arg(accel));
         recentFiles[index]->setToolTip(files[index]); // set the full name that we need later for saving
         recentFiles[index]->setData(QVariant(index));
+        QString accel(tr("none"));
         if (index < shortcut_count){
-            recentFiles[index]->setShortcut(accel);
+            auto accel_tmp = QString::fromStdString(shortcut_modifiers);
+            accel_tmp.append(QString::number(index+1,10)).toStdString();
+            auto check = Application::Instance->commandManager().checkAcceleratorForConflicts(qPrintable(accel_tmp));
+            if (check) {
+                recentFiles[index]->setShortcut(QKeySequence());
+                auto msg = QLatin1String("Recent macros : shortcut %1 disabled because conflicting with %2\n")
+                                                            .arg(accel_tmp).arg(QLatin1String(check->getName()));
+                Base::Console().Warning(qPrintable(msg));

```

Comment:
DON'T do this! For more details see: https://owasp.org/www-community/attacks/Format_string_attack
Instead write:
`Base::Console().Warning("%s", qPrintable(msg));`

Reply:
Fixed. Thanks.